src/pkg/services/m365/api/drive.go

```diff
@@ -305,3 +306,13 @@ func (c Drives) DeleteItemPermission(
 
 	return nil
 }
+
+// DriveItemCollisionKeyy constructs a key from the item name.
+// collision keys are used to identify duplicate item conflicts for handling advanced restoration config.
+func DriveItemCollisionKey(item models.DriveItemable) string {
+	if item == nil {
+		return ""
+	}
+
+	return ptr.Val(item.GetName())
```

Comment:
is the cache going to be per drive or per folder? If it's per drive the collision key will need to include the full (display name) path of the item

Reply:
It will be per folder.